contracts/DxMgnPool.sol

```diff
@@ -82,9 +85,29 @@ contract DxMgnPool {
         uint auctionIndex = dx.getAuctionIndex(address(depositToken), address(secondaryToken));
         require(auctionIndex > lastParticipatedAuctionIndex, "Has to wait for new auction to start");
 
-        // ... super call into trader to participate in next auction
+        address sellToken;
+        address buyToken;
+        if (isPrimiaryTokenTurn){
+            //depositng new tokens
+            uint amount = depositToken.balanceOf(address(this));
+            if(amount > 0) {
+                require(depositToken.approve(address(dx), amount), "Token transfer from Pool to dx was not approved");
+                dx.deposit(address(depositToken), amount);
+            }
+
+            sellToken = address(secondaryToken);
+            buyToken = address(depositToken);
+        }
+        else {
+            sellToken = address(depositToken);
+            buyToken = address(secondaryToken); 
+        }
+        if (lastParticipatedAuctionIndex != 0)
+            dx.claimSellerFunds(buyToken, sellToken, address(this), lastParticipatedAuctionIndex);
+        uint amount = dx.balances(address(sellToken), address(this));
+        (lastParticipatedAuctionIndex, ) = dx.postSellOrder(sellToken, buyToken, 0, amount);
```

Comment:
`lastParticipatedAuctionIndex` above is computed always from depositToken -> secondaryToken. We should probably make it the same here, otherwise we might get inconsistencies.